src/cosmos-keystore.ts

```diff
@@ -77,8 +77,20 @@ export function testPassword(address: string, password: string) {
 }
 
 // returns the index of the stored wallets
-export function getWalletIndex(): [{ name: string; address: string }] {
-  return JSON.parse(localStorage.getItem(KEY_TAG + '-index') || '[]')
+export function getWalletIndex(enriched:Boolean = true): WalletIndex[] {  
```

Comment:
why this switch? why not always return the enriched data?

Reply:
It's used to return the data if writing functions, so it it will always return this data, this function become useless after second call, cause network data will be stored in wallet index in localStorage, so in this case, a better approach will be adding network in addtoindex function, but then, to many fields will be duplicated in index and in wallet. Doesn't sound good.